src/middlewares/user.middleware.js

```diff
@@ -54,3 +57,51 @@ export const checkValidOldPassword = async (req, res, next) => {
   }
   next();
 };
+export const checkValidPassword = async (req, res, next) => {
+	const { password } = req.body;
+	const user  = req.user;
+	const isValidPassword = await BcryptUtility.verifyPassword(
+		password,
+		user.password
+	);
+	if (!isValidPassword)
+		return res.status(400).json({ message: 'Wrong credentials, try again.' });
+	next();
+}
+
+export const verifyUserEmail = async(req, res, next) =>{
+    const userToken = req.params.token;
+    const getToken = await jwtTokens.findOne({
+      where: { token: userToken},
+      });
+      if(getToken && !getToken.revoked){
+        const decoded = JwtUtility.verifyToken(userToken);
+        if(decoded){
+        await jwtTokens.update({ revoked: true }, {
+            where: { token: userToken }
+        });
+        const getUser = await User.findByPk(decoded.id);
+        req.user = getUser;
+        next();
+    } else{
+        return res.status(403).json({message: 'Failed to to verify email'});
+    }
+    } else {
+        return res.status(403).json({message: 'Failed to to verify email'});
+      }
+  }
```

Comment:
Let's split this middleware into the following:
- Verify if the token is not revoked
- Use data provided in the above middleware to implement verify email

Reply:
done